### ISSUE TITLE
AttributeError

### ISSUE THREAD

**Issue by Developer-Alpha** [explosion/spaCy#1163]
Attribute error arises when I provide the path to output_dir

```
"""A quick example for training a part-of-speech tagger, without worrying
about the tokenization, or other language-specific customizations."""

from __future__ import unicode_literals
from __future__ import print_function

import plac
from pathlib import Path

from spacy.vocab import Vocab
from spacy.tagger import Tagger
from spacy.tokens import Doc
from spacy.gold import GoldParse

import random

# You need to define a mapping from your data's part-of-speech tag names to the
# Universal Part-of-Speech tag set, as spaCy includes an enum of these tags.
# See here for the Universal Tag Set:
# http://universaldependencies.github.io/docs/u/pos/index.html
# You may also specify morphological features for your tags, from the universal
# scheme.
TAG_MAP = {
    'N': {"pos": "NOUN"},
    'V': {"pos": "VERB"},
    'J': {"pos": "ADJ"}
}

# Usually you'll read this in, of course. Data formats vary.
# Ensure your strings are unicode.
DATA = [
    (
        ["ids", "id's", "invoiceid", "invoice_id", "Rajeev"],
        ["N", "N", "N", "N", "N"]
    )
]


def ensure_dir(path):
    if not path.exists():
        path.mkdir()


def main(output_dir=None):
    output_dir = '/home/adish/tag1'
    if output_dir is not None:
        output_dir = Path(output_dir)
        ensure_dir(output_dir)
        ensure_dir(output_dir / "pos")
        ensure_dir(output_dir / "vocab")

    vocab = Vocab(tag_map=TAG_MAP)
    # The default_templates argument is where features are specified. See
    # spacy/tagger.pyx for the defaults.
    tagger = Tagger(vocab)
    for i in range(25):
        for words, tags in DATA:
            doc = Doc(vocab, words=words)
            gold = GoldParse(doc, tags=tags)
            tagger.update(doc, gold)
        random.shuffle(DATA)
    tagger.model.end_training()
    doc = Doc(vocab, orths_and_spaces=zip(
        ["ID", "Id", "iD", "IDs", "id", "ids", "id's", "ID's", "invoice id", "inv id", "Rajeev"], [True] * 11))
    tagger(doc)
    for word in doc:
        print(word.text, word.tag_, word.pos_)
    if output_dir is not None:
        tagger.model.dump(str(output_dir / 'pos' / 'model'))
        with (output_dir / 'vocab' / 'strings.json').open('w') as file_:
            tagger.vocab.strings.dump(file_)


if __name__ == '__main__':
    plac.call(main)
```

The following error arises:- 

```
/usr/bin/python2.7 /home/adish/spaCy-2.0.0-alpha/examples/training/train_tagger.py
ID N NOUN
Id N NOUN
iD N NOUN
IDs N NOUN
id N NOUN
ids N NOUN
id's N NOUN
ID's N NOUN
invoice id N NOUN
inv id N NOUN
Rajeev N NOUN
Traceback (most recent call last):
  File "/home/adish/spaCy-2.0.0-alpha/examples/training/train_tagger.py", line 75, in <module>
    plac.call(main)
  File "/usr/local/lib/python2.7/dist-packages/plac_core.py", line 328, in call
    cmd, result = parser.consume(arglist)
  File "/usr/local/lib/python2.7/dist-packages/plac_core.py", line 207, in consume
    return cmd, self.func(*(args + varargs + extraopts), **kwargs)
  File "/home/adish/spaCy-2.0.0-alpha/examples/training/train_tagger.py", line 71, in main
    tagger.vocab.strings.dump(file_)
AttributeError: 'spacy.strings.StringStore' object has no attribute 'dump'

Process finished with exit code 1


```
Can anyone provide any information regarding this issue ?

**Comment by ines** [comment]
From your code, it looks like you are using spaCy v2.0 alpha? If so, this is likely the problem: In v2, all serialization methods have been updated to use a consistent API ([see here](http://alpha.spacy.io/docs/usage/v2#incompat) for the comparison).

So `StringStore.dump` has been replaced with `StringStore.to_disk` and `StringStore.to_bytes`. I'm also not 100% sure if the example above already works as expected with v2.0 alpha – if you run into more problems, you might have to go back to using the stable v1.x for now.

**Comment by Developer-Alpha** [comment]
That works.. But the JSON file that is being generated in the VOCAB folder only contains the data that I wanted to train and that too in random order. I want to save and load that tagger in my main code. Will that work ?

**Comment by ines** [comment]
We just released the new version `v2.0.0a18` and also updated the tagger training example: https://github.com/explosion/spaCy/blob/develop/examples/training/train_tagger.py

This should hopefully give you everything you need. I've also explained some more about training the tagger in v1.x vs 2.0 in #1483.
